### PR TITLE
Code coach modularize

### PR DESCRIPTION
**Pull request by ascholerChemeketa**
- Moved Pyflakes code coach to its own file.
- Update coach run/render code to more cleanly support multiple coaches. 
- Set up Code Coach string for internationalization.
- Fix bug where coach div not removed if all issues fixed

Coaches are added to a list during **ActiveCode** construction.

**runCoaches** function is triggered on program run. It asynchronously runs all coaches, but then waits for all results so Code Coach div can display any coach feedback deterministically in the order coaches were created. 

Questions/Comments:
- I could definitely see a case for more flexible addition of coaches to ActiveCodes instead of hard coding based on language in the constructor. I just don't have a clear idea of what that should look like.
- I *think* the internationalization is right...

### CODE REVIEW

**Comment by ascholerChemeketa** [comment]
I stress-tested the deterministic output with a **TestCoach** that just waited a random amount of time before returning a set string. Works as intended when there are multiple coaches taking variable amounts of time.

**Comment by ascholerChemeketa** [comment]
One aspect that might be annoying in existing books - Pyflakes complains about * imports.

Personally, I am using this as an excuse to replace them all in my book. But it would not be too hard to filter them out in the Pyflakes coach.

image

File: runestone/activecode/js/activecode.js [hunk 1]
```diff
@@ -1222,6 +1227,35 @@ Yet another is that there is an internal error.  The internal error message is:
...
+
+        //clear anything after header in codecoach div and hide it
+        $(this.codecoach).children().slice(1).remove();
+        $(this.codecoach).css("display", "none");
+
+        //get code, run coaches
+        let code = await this.buildProg(false);
+        let results = [];
+        for(let coach of this.codeCoachList) {
+            results.push(coach.check(code));
```
Comment (bnmnetp) [inline runestone/activecode/js/activecode.js:1241]: Should there be a method to add a coach?  If I'm thinking about the future and adding a new coach I would want to have a method to do that so I don't have to modify this code...

Comment (ascholerChemeketa) [reply runestone/activecode/js/activecode.js:1241]: No one should have to touch that code. 

Coaches are added up in the constructor (line 143). I'm not sure where else they would be injected from - that was one of my questions. I don't have a good sense of how to make creating coaches more modular.

File: runestone/activecode/js/activecode.js [hunk 2]
```diff
@@ -137,6 +139,12 @@ export class ActiveCode extends RunestoneBase {
             this.caption = "ActiveCode";
         }
         this.addCaption("runestone");
+
+        //Setup CodeCoaches - add based on language
```
Comment (ascholerChemeketa) [inline runestone/activecode/js/activecode.js:143]: Coach creation

File: runestone/activecode/js/activecode.js (hunk 1 above)
Comment (ascholerChemeketa) [reply runestone/activecode/js/activecode.js:1241]: Hrm, can't figure out how to quote in those lines without starting a new conversation.

Comment (bnmnetp) [reply runestone/activecode/js/activecode.js:1241]: I was imagining a method on the Activecode instance, but then I'm not sure what other object would call it, so having it in the constructor is fine for now.

**Comment by ascholerChemeketa** [comment]
Updated to filter import * related warnings from pyflakes to avoid scary warnings in existing books.

**Pull request merged**